solutions/src/main/java/ru/sberbank/school/task10/MyScalableThreadPool.java

```diff
@@ -0,0 +1,189 @@
+        }
+
+        this.sizeMin = sizeMin;
+        this.sizeMax = sizeMax;
+        this.activeThreads = 0;
+        fillPoolMap(sizeMin);
+    }
+
+    public MyScalableThreadPool(int fixedSize) {
+        if (fixedSize <= 0) {
+            throw new IllegalArgumentException("Incorrect input size");
+        }
+        this.sizeMin = fixedSize;
+        this.sizeMax = fixedSize;
+        this.activeThreads = 0;
+        fillPoolMap(fixedSize);
+    }
+
+    private void fillPoolMap(int size) {
+        IntStream.range(0, size)
+                .forEach(i -> poolMap.put(i, new ThreadWorker("ThreadWorker " + i, i)));
+
+        IntStream.range(size, sizeMax)
+                .forEach(freeIndexForNewWorker::add);
+    }
+
+    @Override
+    public void start() {
+        poolMap.forEach((key, worker) -> {
+            worker.start();
+
+            synchronized (activeThreadsLock) {
+                activeThreads++;
+            }
+        });
+    }
+
+    @Override
+    public void stopNow() {
+        synchronized (poolMap) {
+            poolMap.forEach((key, worker) -> worker.interrupt());
+            poolMap.clear();
+            freeIndexForNewWorker.clear();
+        }
+        synchronized (tasks) {
+            tasks.clear();
+        }
+        activeThreads = 0;     //  before tasks.clear() sometimes "activeThreads = 1"
+                               // in last reading... after "= 0" writing    NB!!! repeat hb rule !!!
+    }
+
+    @Override
+    public void execute (Runnable runnable){
+        synchronized (tasks) {
+            tasks.add(new FutureTask(runnable, true));
+            tasks.notifyAll();
+        }
+    }
+
+    @Override
+    public <T> Future <T> execute(Callable <T> callable) {
+        synchronized (tasks) {
+            FutureTask<T> futureTask = new FutureTask(callable);
+            tasks.add(futureTask);
+            tasks.notifyAll();
+            return futureTask;
+        }
+    }
+
+    /*
+    *return true if needed to interrupt, otherwise return false
+    *  */
+    private boolean poolResize () {
+        int taskSize = tasks.size();
+
+        synchronized (poolMap) {
+            poolMap.forEach((key, worker) -> {
+
+                if (worker == null || worker.isInterrupted()) {
+                    worker = null;
+                    if (!freeIndexForNewWorker.contains(key)) {
+                        freeIndexForNewWorker.offer(key);
+                    }
+
+                    synchronized (activeThreadsLock) {
+                        activeThreads--;
+                    }
+                }
+            });
+
+            if (taskSize > sizeMin && activeThreads < sizeMax) {
+                synchronized (activeThreadsLock) {
+                    activeThreads++;
+                }
+
+                if (!freeIndexForNewWorker.isEmpty()) {
+                    int id = freeIndexForNewWorker.poll();
+
+                    ThreadWorker worker =new ThreadWorker("ThreadWorker " + id, id);
```

Comment:
WhitespaceAround: &#39;&#61;&#39; is not followed by whitespace.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1516499717>*